src/nvim/eval/typval_encode.c.h

```diff
@@ -741,6 +741,7 @@ TYPVAL_ENCODE_SCOPE int _TYPVAL_ENCODE_ENCODE(
       case kMPConvPartial: {
         partial_T *const pt = cur_mpsv->data.p.pt;
         tv = cur_mpsv->tv;
+        (void)tv;
         switch (cur_mpsv->data.p.stage) {
           case kMPConvPartialArgs: {
             TYPVAL_ENCODE_CONV_FUNC_BEFORE_ARGS(tv,
```

Comment:
@ZyX-I seems like `TYPVAL_ENCODE_CONV_FUNC_BEFORE_ARGS` et. al don't use `tv` param, but I assume it's needed for some reason.

Reply:
Many of those `tv` parameters are there for consistency reasons: better have and not need then not have and refactor 100500 macros definitions when next encoder appears to need them (or, rather, not them, but some other data extracted from `tv`: many of the macros accept a pair of `tv` and preextracted data). Also AFAIR they allowed putting debugging prints which is better reason for their existence as some macros simply receive `NULL` as `tv` and some are unlikely to need `tv` as you e.g. would not really need it to dump what is between list items: there hardly is any format which will require writing something like `{'foo', [2]'bar', [2]'baz'}` (where `foo` goes without array length and `bar` goes with: what else could you use `tv` for in position where you dump commas given that said `tv` points to list itself and not list entry?).